gdc/docker.go

```diff
@@ -107,33 +107,78 @@ func mainCommand(compose ComposeInfo) string {
 
 func executeDockerCommand(compose ComposeInfo, service string, command string, inputFile string) {
 	if len(inputFile) > 0 {
-		RunCommands(
+		cmds := []string{
 			fmt.Sprintf("cat %s", inputFile),
 			fmt.Sprintf("%s exec -T %s %s", mainCommand(compose), service, command),
-		)
+		}
+		err := RunCommands(cmds...)
+		if err != nil {
+			Exit("Error running command! %v", cmds)
+		}
 	} else {
 		RunCommand("%s exec %s %s", mainCommand(compose), service, command)
 
 	}
 }
 
 func ecrLogin() {
-	RunCommands("aws ecr get-login-password", "docker login --password-stdin -u AWS 421990735784.dkr.ecr.us-east-1.amazonaws.com")
+	cmds := []string{
+		"aws ecr get-login-password",
+		"docker login --password-stdin -u AWS 421990735784.dkr.ecr.us-east-1.amazonaws.com",
+	}
+	err := RunCommands(cmds...)
+	if err != nil {
+		Exit("Error logging into ECR! %v", err)
+	}
+}
+
+func ghcrLogin() {
+	// Try gh CLI first - get username
+	username, err := RunCommandsOutput("gh api user -q .login")
+	if err == nil && username != "" {
+		fmt.Println("Logging into GitHub Container Registry using GitHub CLI credentials...")
+		err = RunCommands("gh auth token", fmt.Sprintf("docker login ghcr.io -u %s --password-stdin", username))
+		if err == nil {
+			return
+		}
+	}
+
+	// Try environment variables
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	githubUsername := os.Getenv("GITHUB_USERNAME")
+	if githubToken != "" && githubUsername != "" {
+		fmt.Println("Logging into GitHub Container Registry using environment variables...")
+		err = RunCommands(
+			fmt.Sprintf("echo %s", githubToken),
+			fmt.Sprintf("docker login ghcr.io -u %s --password-stdin", githubUsername),
+		)
+		if err == nil {
+			return
+		}
+	}
+
+	fmt.Println("Warning: Could not login to ghcr.io. Continuing anyway...")
```

Comment:
Should this be a fatal error? If `login-to-ghcr` was passed, it should be a failure if you couldn't log in.